### PR TITLE
Zipped pbcore to AAPB

### PR DESCRIPTION
**Pull request by foglabs**

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,2 +1,19 @@
 module ApplicationHelper
+
+  def split_and_validate_ids(input)
+    return false unless input =~ /[a-z0-9\-_\/\n]/
+    ids = input.split(/\s/).reject(&:empty?)
+    return false unless ids.all? {|id| /cpb-aacip[\-_\/][0-9]{1,3}[\-_\/][a-zA-Z0-9]/ =~ id }
```
Comment (afred) [inline app/helpers/application_helper.rb]: the regex here might accidentally match if an errant non-alphanumeric character accidentally get's added to the end of one of the ids, or if characters accidentally get added to the beginning. E.g..
```
/cpb-aacip[\-_\/][0-9]{1,3}[\-_\/][a-zA-Z0-9]/ =~ "cpb-aacip-123-45^789"
# => 0
/cpb-aacip[\-_\/][0-9]{1,3}[\-_\/][a-zA-Z0-9]/ =~ "whoops-cpb-aacip-123-45789"
# => 7
```
Using anchors to match beginning and end of string, and adding a `+` for > 1 alphanum should work...
```
/\Acpb-aacip[\-_\/][0-9]{1,3}[\-_\/][a-zA-Z0-9]+\z/ =~ "cpb-aacip-123-457^89"
# => nil
/\Acpb-aacip[\-_\/][0-9]{1,3}[\-_\/][a-zA-Z0-9]+\z/ =~ "whoops-cpb-aacip-123-457^89"
# => nil
/\Acpb-aacip[\-_\/][0-9]{1,3}[\-_\/][a-zA-Z0-9]+\z/ =~ "cpb-aacip-123-45789"
# => 0 
```

File: app/models/ams/pbcore_xml_export_extension.rb [hunk 2]
```diff
@@ -60,31 +62,25 @@ def prepare_asset(xml)
...
-    # Producing Org
-    producing_organization.to_a.each do |org|
-      xml.pbcoreCreator do |creator_node|
-        creator_node.creator { creator_node.text(org) }
-        creator_node.creatorRole { creator_node.text('Producing Organization') }
-      end
-    end
+    # no pbcoreRelation
+      # no pbcoreRelationType
+      # no pbcoreRelationIdentifier
```
Comment (afred) [inline app/models/ams/pbcore_xml_export_extension.rb]: remove these comments?

File: app/controllers/catalog_controller.rb [hunk 3]
```diff
@@ -482,6 +482,71 @@ def render_bookmarks_control?
...
+  #   query_params.delete :page
+  #   query_params.delete :per_page
+  #   query_params.delete :action
+  #   query_params.delete :controller
+  #   query_params.delete :locale
+  #   ExportRecordsJob.perform_later(query_params, current_user)
+  #   flash[:notice] = "Your IDs have been accepted."
+  #   render 'pb_to_aapb_form'
+  # end
+
```
Comment (afred) [inline app/controllers/catalog_controller.rb]: remove this commented out stuff?

File: app/services/ams/export/documents_to_zipped_pbcore.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+require 'zip'
+#NEW NEW NEW
```
Comment (afred) [inline app/services/ams/export/documents_to_zipped_pbcore.rb]: ☝️remove?

File: app/services/aapb/batch_ingest/pbcore_xml_mapper.rb [hunk 5]
```diff
@@ -48,7 +48,8 @@ def asset_attributes
           # pull out no-type titles, removing from grouped_titles
           titles_no_type = grouped_titles.slice!(*title_types)
           attrs[:title]                       = titles_no_type.values.flatten
-          attrs[:episode_title]               = (grouped_titles.fetch("episode title", []) + grouped_titles.fetch("episode", []))
+          # attrs[:episode_title]               = (grouped_titles.fetch("episode title", []) + grouped_titles.fetch("episode", []))
```
Comment (afred) [inline app/services/aapb/batch_ingest/pbcore_xml_mapper.rb]: ☝️ remove?

File: app/services/ams/export/export_service.rb [hunk 6]
```diff
@@ -47,6 +47,24 @@ def upload_to_s3
...
+        filepath = @file_path.path
+        filename = File.basename(@file_path.path)
+        aapb_host = 'ec2-18-213-230-80.compute-1.amazonaws.com'
+        # config aapb host
+        if aapb_key_path && aapb_host.present? && filepath.present?
+
+          `scp #{aapb_key_path} #{filepath} ec2-user@#{aapb_host}:/home/ec2-user/ingest_zips/#{filename}`
+          output =  `ssh -t #{aapb_key_path} ec2-user@#{aapb_host} 'cd /home/ec2-user/ingest_zips && unzip #{filename} && cd /var/www/aapb/current && RAILS_ENV=production /usr/bin/ruby scripts/download_clean_ingest.rb --stdout-log --files /home/ec2-user/ingest_zips/*.xml'`
+          puts output
+          Sidekiq::Logging.logger.info output
```
Comment (afred) [inline app/services/ams/export/export_service.rb]: I think we'll probably just want `Rails.logger.info` here

File: app/services/ams/export/export_service.rb [hunk 7]
```diff
@@ -47,6 +47,24 @@ def upload_to_s3
         end
         @s3_path = obj.public_url
       end
+
+      def scp_to_aapb(user)
+        # aapb_key_path = '-i /home/ec2-user/.ssh/aapb-zip-key.pem'
+        aapb_key_path = ""
+        filepath = @file_path.path
+        filename = File.basename(@file_path.path)
+        aapb_host = 'ec2-18-213-230-80.compute-1.amazonaws.com'
```
Comment (afred) [inline app/services/ams/export/export_service.rb]: change the hardcoded host to `ENV['AAPB_HOST']`?

File: app/services/ams/export/export_service.rb (hunk 6 above)
Comment (afred) [reply app/services/ams/export/export_service.rb]: Also, we should provide some useful error messages if the `ENV['AAPB_HOST']` is not set, or if the AAPB machine is not reachable.

**Comment by foglabs** [comment]
Need to add AAPB_HOST to travis

**Pull request merged**